Assets/Scripts/Game/MagicAndEffects/EntityEffectManager.cs

```diff
@@ -2131,6 +2131,10 @@ private void PlayerSpellCasting_OnReleaseFrame()
                 lastSpell = readySpell;
                 lastReadySpellCastingCost = readySpellCastingCost;
             }
+            else
+            {
+                lastReadySpellCastingCost = 0;
+            }
```

Comment:
Can we flip this **if/else** so it would check the positive condition? It's just a personal preference, feel free to ignore this comment

```cs
            if (readySpellDoesNotCostSpellPoints)
            {
                lastReadySpellCastingCost = 0;
            }
            else
            {
                lastSpell = readySpell;
                lastReadySpellCastingCost = readySpellCastingCost;
            }
```

Reply:
Done, but that barely improves the code readability, I also considered handling lastReadySpellCastingCost separately from lastSpell because I don't really consider their lifecycles to be related...
```
if (!readySpellDoesNotCostSpellPoints)
{
    lastSpell = readySpell;
}
lastReadySpellCastingCost = readySpellDoesNotCostSpellPoints ? 0 : readySpellCastingCost;
```

I wonder if that code could have used less state, and less separate scalar variables to represent that state...
But submitted the smallest change that fixes the bug